include/CppUTest/JUnitTestOutput.h

```diff
@@ -74,6 +74,9 @@ class JUnitTestOutput: public TestOutput
     virtual void writeFailure(JUnitTestCaseResultNode* node);
     virtual void writeFileEnding();
 
+private:
+    ConsoleTestOutput* currentConsoleOutput_;
+    ConsoleTestOutput defaultConsoleOutput_;
```

Comment:
@basvodde I have to admit I think your solution is more general and thus, better. I was wondering though, whether using a default output might be advantageous (that would make two for CompositeTestOutput)

1. new and delete would alway happen where they're supposed to
2. No need to test whether currentOtuptu is NULL because it won't be

The default would probably be StringBufferTestOutput.